modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -125,12 +130,17 @@ func (h *handler) checkDependency(bd *fleet.BundleDeployment) (string, bool, err
 		if err != nil {
 			return "", false, err
 		}
+
+		if len(bds) == 0 {
+			return fmt.Sprintf("no bundles matching labels %s in namespace %s", selector.String(), bundleNamespace), false, nil
+		}
+
 		for _, bd := range bds {
 			c := condition.Cond("Ready")
 			if c.IsTrue(bd) {
 				continue
 			} else {
-				return fmt.Sprintf("%s/%s", bundleNamespace, depend.Name), false, nil
+				return fmt.Sprintf("dependent bundles matching labels %s in namespace %s not ready", selector.String(), bundleNamespace), false, nil
```

Comment:
Let's collect errors and return an error slice instead

Reply:
i have collected the dependent bundleNames and returned them, allowing a custom error that wraps around all dependent bundle names.